distributed/preloading.py

```diff
@@ -0,0 +1,87 @@
+import atexit
+import shutil
+import logging
+import os
+import sys
+from importlib import import_module
+
+from .compatibility import reload, invalidate_caches, cache_from_source
+
+
+logger = logging.getLogger(__name__)
+
+
+def load_file(path):
```

Comment:
It looks like the `upload_file` method depends on this function in `preloading.py`.  Because this function is more general than just the preloading feature perhaps we should instead move it to utils.py?